app/assets/v2/js/pages/bounty_details.js

```diff
@@ -759,19 +759,35 @@ var do_actions = function(result) {
     actions.push(_entry);
   }
 
+  function get_repository_name(github_url) {
+    const repo_link = document.createElement('a');
```

Comment:
why is it necessary to create a new element on the DOM for htis?

Reply:
This is leveraging the DOM's API to create an object with the attributes of a link, making it straightforward to resolve the site's name for non-github bounties. And as nothing is being inserted into the document, this shouldn't negatively impact client performance. As with most things in JavaScript, there are many ways to address this problem. Can you elaborate more as to your questioning of the necessity of this particular approach?

Reply:
👍